vsintegration/src/FSharp.Editor/Diagnostics/DocumentDiagnosticAnalyzer.fs

```diff
@@ -106,26 +104,19 @@ type internal FSharpDocumentDiagnosticAnalyzer
 
     interface IFSharpDocumentDiagnosticAnalyzer with
 
-        member this.AnalyzeSyntaxAsync(document: Document, cancellationToken: CancellationToken): Task<ImmutableArray<Diagnostic>> =
-            if document.Project.IsFSharpMetadata then Task.FromResult(ImmutableArray.Empty)
+        member _.AnalyzeSyntaxAsync(document: Document, cancellationToken: CancellationToken): Task<ImmutableArray<Diagnostic>> =
+            if document.Project.IsFSharpMetadata then
+                Task.FromResult(ImmutableArray.Empty)
             else
+                backgroundTask {
```

Comment:
`backgroundTask` with just a `return!` is redundant. `GetDiagnostics` already returns a `backgroundTask`.